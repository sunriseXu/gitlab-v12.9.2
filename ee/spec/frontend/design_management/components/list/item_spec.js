import { createLocalVue, shallowMount } from '@vue/test-utils';
import VueRouter from 'vue-router';
import Item from 'ee/design_management/components/list/item.vue';

const localVue = createLocalVue();
localVue.use(VueRouter);
const router = new VueRouter();

// Referenced from: doc/api/graphql/reference/gitlab_schema.graphql:DesignVersionEvent
const DESIGN_VERSION_EVENT = {
  CREATION: 'CREATION',
  DELETION: 'DELETION',
  MODIFICATION: 'MODIFICATION',
  NO_CHANGE: 'NONE',
};

describe('Design management list item component', () => {
  let wrapper;
  function createComponent({
    notesCount = 0,
    event = DESIGN_VERSION_EVENT.NO_CHANGE,
    isLoading = false,
  } = {}) {
    wrapper = shallowMount(Item, {
      localVue,
      router,
      propsData: {
        id: 1,
        filename: 'test',
        image: 'http://via.placeholder.com/300',
        isLoading,
        event,
        notesCount,
        updatedAt: '01-01-2019',
      },
      stubs: ['router-link'],
    });
  }

  afterEach(() => {
    wrapper.destroy();
  });

  describe('with notes', () => {
    it('renders item with single comment', () => {
      createComponent({ notesCount: 1 });

      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders item with multiple comments', () => {
      createComponent({ notesCount: 2 });

      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('with no notes', () => {
    it('renders item with no status icon for none event', () => {
      createComponent();

      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders item with correct status icon for modification event', () => {
      createComponent({ event: DESIGN_VERSION_EVENT.MODIFICATION });

      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders item with correct status icon for deletion event', () => {
      createComponent({ event: DESIGN_VERSION_EVENT.DELETION });

      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders item with correct status icon for creation event', () => {
      createComponent({ event: DESIGN_VERSION_EVENT.CREATION });

      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders loading spinner when isLoading is true', () => {
      createComponent({ isLoading: true });

      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
