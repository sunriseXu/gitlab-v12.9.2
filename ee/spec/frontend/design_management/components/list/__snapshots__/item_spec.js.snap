// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Design management list item component with no notes renders item with correct status icon for creation event 1`] = `
<router-link-stub
  class="card cursor-pointer text-plain js-design-list-item design-list-item"
  to="[object Object]"
>
  <div
    class="card-body p-0 d-flex-center overflow-hidden position-relative"
  >
    <div
      class="design-event position-absolute"
    >
      <span
        aria-label="Added in this version"
        title="Added in this version"
      >
        <icon-stub
          class="text-success-500"
          name="file-addition-solid"
          size="18"
        />
      </span>
    </div>
     
    <img
      alt="test"
      class="block ml-auto mr-auto mw-100 mh-100 design-img"
      data-qa-selector="design_image"
      src="http://via.placeholder.com/300"
    />
  </div>
   
  <div
    class="card-footer d-flex w-100"
  >
    <div
      class="d-flex flex-column str-truncated-100"
    >
      <span
        class="bold str-truncated-100"
        data-qa-selector="design_file_name"
      >
        test
      </span>
       
      <span
        class="str-truncated-100"
      >
        
        Updated 
        <timeago-stub
          cssclass=""
          time="01-01-2019"
          tooltipplacement="bottom"
        />
      </span>
    </div>
     
    <!---->
  </div>
</router-link-stub>
`;

exports[`Design management list item component with no notes renders item with correct status icon for deletion event 1`] = `
<router-link-stub
  class="card cursor-pointer text-plain js-design-list-item design-list-item"
  to="[object Object]"
>
  <div
    class="card-body p-0 d-flex-center overflow-hidden position-relative"
  >
    <div
      class="design-event position-absolute"
    >
      <span
        aria-label="Deleted in this version"
        title="Deleted in this version"
      >
        <icon-stub
          class="text-danger-500"
          name="file-deletion-solid"
          size="18"
        />
      </span>
    </div>
     
    <img
      alt="test"
      class="block ml-auto mr-auto mw-100 mh-100 design-img"
      data-qa-selector="design_image"
      src="http://via.placeholder.com/300"
    />
  </div>
   
  <div
    class="card-footer d-flex w-100"
  >
    <div
      class="d-flex flex-column str-truncated-100"
    >
      <span
        class="bold str-truncated-100"
        data-qa-selector="design_file_name"
      >
        test
      </span>
       
      <span
        class="str-truncated-100"
      >
        
        Updated 
        <timeago-stub
          cssclass=""
          time="01-01-2019"
          tooltipplacement="bottom"
        />
      </span>
    </div>
     
    <!---->
  </div>
</router-link-stub>
`;

exports[`Design management list item component with no notes renders item with correct status icon for modification event 1`] = `
<router-link-stub
  class="card cursor-pointer text-plain js-design-list-item design-list-item"
  to="[object Object]"
>
  <div
    class="card-body p-0 d-flex-center overflow-hidden position-relative"
  >
    <div
      class="design-event position-absolute"
    >
      <span
        aria-label="Modified in this version"
        title="Modified in this version"
      >
        <icon-stub
          class="text-primary-500"
          name="file-modified-solid"
          size="18"
        />
      </span>
    </div>
     
    <img
      alt="test"
      class="block ml-auto mr-auto mw-100 mh-100 design-img"
      data-qa-selector="design_image"
      src="http://via.placeholder.com/300"
    />
  </div>
   
  <div
    class="card-footer d-flex w-100"
  >
    <div
      class="d-flex flex-column str-truncated-100"
    >
      <span
        class="bold str-truncated-100"
        data-qa-selector="design_file_name"
      >
        test
      </span>
       
      <span
        class="str-truncated-100"
      >
        
        Updated 
        <timeago-stub
          cssclass=""
          time="01-01-2019"
          tooltipplacement="bottom"
        />
      </span>
    </div>
     
    <!---->
  </div>
</router-link-stub>
`;

exports[`Design management list item component with no notes renders item with no status icon for none event 1`] = `
<router-link-stub
  class="card cursor-pointer text-plain js-design-list-item design-list-item"
  to="[object Object]"
>
  <div
    class="card-body p-0 d-flex-center overflow-hidden position-relative"
  >
    <!---->
     
    <img
      alt="test"
      class="block ml-auto mr-auto mw-100 mh-100 design-img"
      data-qa-selector="design_image"
      src="http://via.placeholder.com/300"
    />
  </div>
   
  <div
    class="card-footer d-flex w-100"
  >
    <div
      class="d-flex flex-column str-truncated-100"
    >
      <span
        class="bold str-truncated-100"
        data-qa-selector="design_file_name"
      >
        test
      </span>
       
      <span
        class="str-truncated-100"
      >
        
        Updated 
        <timeago-stub
          cssclass=""
          time="01-01-2019"
          tooltipplacement="bottom"
        />
      </span>
    </div>
     
    <!---->
  </div>
</router-link-stub>
`;

exports[`Design management list item component with no notes renders loading spinner when isLoading is true 1`] = `
<router-link-stub
  class="card cursor-pointer text-plain js-design-list-item design-list-item"
  to="[object Object]"
>
  <div
    class="card-body p-0 d-flex-center overflow-hidden position-relative"
  >
    <!---->
     
    <gl-loading-icon-stub
      color="orange"
      label="Loading"
      size="md"
    />
  </div>
   
  <div
    class="card-footer d-flex w-100"
  >
    <div
      class="d-flex flex-column str-truncated-100"
    >
      <span
        class="bold str-truncated-100"
        data-qa-selector="design_file_name"
      >
        test
      </span>
       
      <span
        class="str-truncated-100"
      >
        
        Updated 
        <timeago-stub
          cssclass=""
          time="01-01-2019"
          tooltipplacement="bottom"
        />
      </span>
    </div>
     
    <!---->
  </div>
</router-link-stub>
`;

exports[`Design management list item component with notes renders item with multiple comments 1`] = `
<router-link-stub
  class="card cursor-pointer text-plain js-design-list-item design-list-item"
  to="[object Object]"
>
  <div
    class="card-body p-0 d-flex-center overflow-hidden position-relative"
  >
    <!---->
     
    <img
      alt="test"
      class="block ml-auto mr-auto mw-100 mh-100 design-img"
      data-qa-selector="design_image"
      src="http://via.placeholder.com/300"
    />
  </div>
   
  <div
    class="card-footer d-flex w-100"
  >
    <div
      class="d-flex flex-column str-truncated-100"
    >
      <span
        class="bold str-truncated-100"
        data-qa-selector="design_file_name"
      >
        test
      </span>
       
      <span
        class="str-truncated-100"
      >
        
        Updated 
        <timeago-stub
          cssclass=""
          time="01-01-2019"
          tooltipplacement="bottom"
        />
      </span>
    </div>
     
    <div
      class="ml-auto d-flex align-items-center text-secondary"
    >
      <icon-stub
        class="ml-1"
        name="comments"
        size="16"
      />
       
      <span
        aria-label="2 comments"
        class="ml-1"
      >
        
        2
      
      </span>
    </div>
  </div>
</router-link-stub>
`;

exports[`Design management list item component with notes renders item with single comment 1`] = `
<router-link-stub
  class="card cursor-pointer text-plain js-design-list-item design-list-item"
  to="[object Object]"
>
  <div
    class="card-body p-0 d-flex-center overflow-hidden position-relative"
  >
    <!---->
     
    <img
      alt="test"
      class="block ml-auto mr-auto mw-100 mh-100 design-img"
      data-qa-selector="design_image"
      src="http://via.placeholder.com/300"
    />
  </div>
   
  <div
    class="card-footer d-flex w-100"
  >
    <div
      class="d-flex flex-column str-truncated-100"
    >
      <span
        class="bold str-truncated-100"
        data-qa-selector="design_file_name"
      >
        test
      </span>
       
      <span
        class="str-truncated-100"
      >
        
        Updated 
        <timeago-stub
          cssclass=""
          time="01-01-2019"
          tooltipplacement="bottom"
        />
      </span>
    </div>
     
    <div
      class="ml-auto d-flex align-items-center text-secondary"
    >
      <icon-stub
        class="ml-1"
        name="comments"
        size="16"
      />
       
      <span
        aria-label="1 comment"
        class="ml-1"
      >
        
        1
      
      </span>
    </div>
  </div>
</router-link-stub>
`;
